### PR TITLE
Modified SAMPLESHEET_CHECK module and INPUT_CHECK subworkflow to take desired samplesheet structure

### PR DESCRIPTION
**Pull request by CarsonJM**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/phageannotator/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/phageannotator _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: assets/samplesheet_full.csv
Comment (maxulysse) [inline assets/samplesheet_full.csv:1]: I'd recommend versioning the samplesheets.
For example you can put them all in a 1.0.0 subfolder

**Comment by CarsonJM** [comment]
Any other recommendations with respect to samplesheet format or INPUT_CHECK/SAMPLESHEET_CHECK?

**Comment by maxulysse** [comment]
We're not at the point where we have global recommendations for the fields in the samplesheet.
But you can have a look at the nf-validation plugin to replace the INPUT_CHECK subworkflow. check the `#nf-validation` channel on slack

**Comment by CarsonJM** [comment]
Thanks for the feedback, is that going to be the recommended method going forward?

**Comment by maxulysse** [comment]
We started doing that in rnaseq and Sarek, so yeah that's the way to go

**Pull request closed without merging**